### PR TITLE
Fixed a bug that was causing an endless loop when fetching a profile picture

### PR DESCRIPTION
**Pull request by ethangraham2001**
Due to our use of `Flow`s during the image-fetching process, our composable functions calling `profileViewModel.fetchImage()` were looping infinitely upon receiving a `404` error from Firebase Storage. Addressed this by:
- Creating a global `ProfilePicture` composable that works as intended, to be called by any other composable that needs it
- using `remember` so that the `fetchImage` flow is only observed when the `profile` passed as arg. changes

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/ui/map/PreviewItineraryScreen.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/map/PreviewItineraryScreen.kt:1]: Lgtm, removed the old ProfilePicture to add the new one

File: app/src/main/java/com/github/wanderwise_inc/app/ui/profile/ProfilePicture.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/profile/ProfilePicture.kt:1]: Good way of doing that. Using remember and calling inside the fetchFunction makes it called whenever it needs to be called. Well done. Good also that now the profile picture is in a separate file

Comment (gunnm0) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/profile/ProfilePicture.kt:1]: elegant fix, doesn't fetch image multiple times

**Review by gunnm0 — APPROVED** [review]
thank you for changing the tests

**Review by engjellismaili — COMMENTED** [review]
Good fix, this was a must TODO since it can cause flooding and would maybe cause other bugs. Well done

**Pull request merged**